src/Simplify.elm

```diff
@@ -4108,6 +4135,54 @@ isNegatableOperator op =
             Nothing
 
 
+basicsToFloatChecks : CheckInfo -> Maybe (Error {})
+basicsToFloatChecks checkInfo =
+    case Evaluate.getNumber checkInfo checkInfo.firstArg of
+        Just _ ->
+            Just
+                (Rule.errorWithFix
+                    { message = "Unnecessary " ++ qualifiedToString (qualify checkInfo.fn defaultQualifyResources) ++ " on a literal number"
+                    , details =
+                        [ "A literal integers is considered as both an Int and a Float, there is therefore no need to explicitly convert it to a Float."
```

Comment:
Typo "integers" → "integer" and maybe mention `number`? Something like

"A literal integer is considered a `number` which means it can be used as both an Int and a Float and there is no need to explicitly convert it to a Float."